extra/innochecksum.cc

```diff
@@ -1428,6 +1453,101 @@ rewrite_checksum(
 		&& !write_file(filename, fil_in, buf, flags, pos);
 }
 
+/** Read and validate page 0, then initialize tablespace flags
+and page size.
+@param	fil_in        File pointer
+@param	buf           Buffer to read page into
+@return whether the page was read successfully */
+static bool read_and_validate_page0(FILE *fil_in, byte *buf)
+{
+  /* Read the minimum page size first */
+  size_t initial_page_size= UNIV_ZIP_SIZE_MIN;
+  if (use_tablespace_flags)
+  {
+    if (!fil_space_t::is_valid_flags(tablespace_flags,
+                                     cur_space != 0))
+    {
+      fprintf(stderr, "Warning: Provided tablespace_flags "
+              "is not valid. So skipping --tablespace_flags option");
+      use_tablespace_flags= false;
+    }
```

Comment:
Can we validate this in `innochecksum_get_one_option()`? This should be a hard error.